components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/tool-editor/views/tool-editor.js

```diff
@@ -176,10 +176,18 @@ define(['require', 'jquery', 'backbone', 'lodash', 'log', 'design_view', "./sour
                                 // The following code has been added to the setTimeout() method because
                                 // the code needs to run asynchronously for the loading screen
                                 setTimeout(function () {
-                                    designView.emptyDesignViewGridContainer();
-                                    designContainer.show();
-                                    designView.renderDesignGrid(self.JSONObject);
-                                    loadingScreen.hide();
+                                    var fileHashCode = application.tabController.getActiveTab().getFile().attributes.hashCode;
+                                    var designViewHash = designView.getHashCode();
+                                    if (fileHashCode != designViewHash || (fileHashCode == undefined && designViewHash == undefined)) {
+                                        designView.setHashCode(fileHashCode);
+                                        designView.emptyDesignViewGridContainer();
+                                        designContainer.show();
+                                        designView.renderDesignGrid(self.JSONObject);
+                                        loadingScreen.hide();
+                                    } else {
```

Comment:
you take out designContainer.show(); and loadingScreen.hide(); and just have the if statement

Reply:
cannot take out the desigContainer.show() out because for the rendering it should be showed first.Then the rendering happened.Can take out the loadingScreen.hide()  but have to put the desigContainer.show() at outside also.When the fresh rendering happens desigContainer.show() calls twice.